### DISCUSSION TITLE
qiime2meco file conversion error QIIME 2 to microtable

### DISCUSSION THREAD

**Discussion by amrvx4** [ChiLiubio/microeco#121] [Q&A] [answered]
Hi Liu,
When I was trying to convert qiime2 (QZA) files to microeco object, I get this error (see below)

qiime2meco(ASV_data = abund_file_path)

Error in initialize(...) : 
Some columns in otu_table are not numeric vector! Please check the otu_table!

I believe there are columns in QIIME 2 (QZA file) which are not numeric. My question: is there possibly anyway to do it within the command line so only numeric columns can be selected or would I have to go back to work on my QIIME2 file first?

Thanks,

**Comment by ChiLiubio** [comment, marked as answer]
Hi @amrvx4 

Generally, feature table is weird if it has some columns not numeric. Please manually check it with the internal command read_qza in file2meco. It can help convert QZA to commonly-used format. An example:
```
# use the file inside file2meco as the example
path <- system.file("extdata", "dada2_table.qza", package="file2meco")
test <- file2meco:::read_qza(path)
View(test$data)
# test is a list
```
After correcting the files, you can manually construct microtable object to skip the qiime2meco function.

Chi
